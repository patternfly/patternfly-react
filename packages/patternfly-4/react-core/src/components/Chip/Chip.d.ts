import { SFC, HTMLProps } from 'react';
import { Omit, OneOf } from '../../typeUtils';
import { TooltipPosition } from '../Tooltip';

export interface ChipProps extends HTMLProps<HTMLDivElement> {
  children?: string;
  closeBtnAriaLabel: string;
  isOverflowChip: boolean;
  tooltipPosition: OneOf<typeof TooltipPosition, keyof typeof TooltipPosition>;
}

declare const Chip: SFC<ChipProps>;

export default Chip;

