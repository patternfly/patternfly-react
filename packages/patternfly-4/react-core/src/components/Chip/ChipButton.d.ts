import { FunctionComponent, HTMLProps } from 'react';

export interface ChipButtonProps extends HTMLProps<HTMLButtonElement> {}

declare const ChipButton: FunctionComponent<ChipButtonProps>;

export default ChipButton;
