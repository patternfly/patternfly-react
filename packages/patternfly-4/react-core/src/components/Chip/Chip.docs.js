import { Chip } from '@patternfly/react-core';
import SimpleExample from './examples/SimpleChip';

export default {
  title: 'Chip',
  components: {
    Chip,
  },
  examples: [
    {
      component: SimpleExample,
      title: 'Simple Example'
    }
  ]
};
