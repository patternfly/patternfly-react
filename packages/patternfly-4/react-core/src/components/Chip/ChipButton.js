import React from 'react';
import { css } from '@patternfly/react-styles';
import PropTypes from 'prop-types';
import styles from '@patternfly/patternfly/components/Chip/chip.css';
import { Button } from '../Button';

const ChipButton = ({ ariaLabel, children, className, onClick, ...props }) => {
  return (
    <Button variant="plain" aria-label={ariaLabel} onClick={onClick} className={className} {...props}>
      {children}
    </Button>
  );
};

ChipButton.propTypes = {
  /** Aria label for chip button */
  ariaLabel: PropTypes.string,
  /** Content rendered inside the chip item */
  children: PropTypes.node,
  /** Additional classes added to the chip item */
  className: PropTypes.string,
  /** Function that is called when clicking on the chip button */
  onClick: PropTypes.func,
};

ChipButton.defaultProps = {
  ariaLabel: 'close',
  children: null,
  className: '',
  onClick: () => { }
};

export default ChipButton;
