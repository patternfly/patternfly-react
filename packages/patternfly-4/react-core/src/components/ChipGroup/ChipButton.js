import React from 'react';
import PropTypes from 'prop-types';
import { Button } from '../Button';

const propTypes = {
  /** Aria label for chip button */
  'aria-label': PropTypes.string,
  /** Content rendered inside the chip item */
  children: PropTypes.node,
  /** Additional classes added to the chip item */
  className: PropTypes.string,
  /** Function that is called when clicking on the chip button */
  onClick: PropTypes.func
};

const defaultProps = {
  'aria-label': 'close',
  children: null,
  className: '',
  onClick: () => { }
};

const ChipButton = ({ children, ...props }) => (
  <Button variant="plain"  {...props}>
    {children}
  </Button>
);

ChipButton.propTypes = propTypes;
ChipButton.defaultProps = defaultProps;

export default ChipButton;
