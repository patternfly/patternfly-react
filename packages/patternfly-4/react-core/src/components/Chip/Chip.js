import React from 'react';
import { css } from '@patternfly/react-styles';
import PropTypes from 'prop-types';
import ChipButton from './ChipButton';
import { Tooltip, TooltipPosition } from '../Tooltip';
import { TimesCircleIcon } from '@patternfly/react-icons';
import styles from '@patternfly/patternfly-next/components/Chip/chip.css';
import GenerateId from '../../internal/GenerateId/GenerateId';
class Chip extends React.Component {
  span = React.createRef();
  state = { isTooltipVisible: false };

  componentDidMount() {
    this.setState({ isTooltipVisible: this.span.current && this.span.current.offsetWidth < this.span.current.scrollWidth });
  }

  renderOverflowChip = () => {
    const { children, className, onClick } = this.props;
    return (
      <div className={css(styles.chip, styles.modifiers.overflow, className)}>
        <ChipButton onClick={onClick}>
          <span className={css(styles.chipText)}>{children}</span>
        </ChipButton>
      </div>
    );
  }

  renderChip = (randomId) => {
    const {
      children,
      closeBtnAriaLabel,
      tooltipPosition,
      className,
      onClick,
    } = this.props;
    if (this.state.isTooltipVisible) {
      return (
        <Tooltip position={tooltipPosition} content={children}>
          <div className={css(styles.chip, className)}>
            <span ref={this.span} className={css(styles.chipText)} id={randomId}>
              {children}
            </span>
            <ChipButton onClick={onClick} ariaLabel={closeBtnAriaLabel} id={`remove_${randomId}`} aria-labelledby={`remove_${randomId} ${randomId}`}>
              <TimesCircleIcon aria-hidden="true" />
            </ChipButton>
          </div>
        </Tooltip>
      )
    } else {
      return (
        <div className={css(styles.chip, className)}>
          <span ref={this.span} className={css(styles.chipText)} id={randomId}>
            {children}
          </span>
          <ChipButton onClick={onClick} ariaLabel={closeBtnAriaLabel} id={`remove_${randomId}`} aria-labelledby={`remove_${randomId} ${randomId}`}>
            <TimesCircleIcon aria-hidden="true" />
          </ChipButton>
        </div>
      )
    }
  }

  render() {
    const {
      isOverflowChip,
    } = this.props;
    return (
      <GenerateId>
        {(randomId) =>
          (
            <React.Fragment>
              {isOverflowChip ? this.renderOverflowChip() : this.renderChip(randomId)}
            </React.Fragment>
          )
        }
      </GenerateId>
    )
  }
}
Chip.propTypes = {
  /** Content rendered inside the chip text */
  children: PropTypes.string,
  /** Aria Label for close button */
  closeBtnAriaLabel: PropTypes.string,
  /** ID of the chip */
  id: PropTypes.string,
  /** Additional classes added to the chip item */
  className: PropTypes.string,
  /** Flag indicating if the chip has overflow*/
  isOverflowChip: PropTypes.bool,
  /** Position of the tooltip which is displayed if text is longer */
  tooltipPosition: PropTypes.oneOf(Object.values(TooltipPosition)),
  /** Function that is called when clicking on the chip button */
  onClick: PropTypes.func,
};

Chip.defaultProps = {
  id: undefined,
  closeBtnAriaLabel: 'close',
  className: '',
  tooltipPosition: 'top',
  isOverflowChip: false,
};

export default Chip;
