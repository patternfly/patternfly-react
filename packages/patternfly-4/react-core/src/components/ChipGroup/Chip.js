import React from 'react';
import { css } from '@patternfly/react-styles';
import PropTypes from 'prop-types';
import ChipButton from './ChipButton';
import { Tooltip, TooltipPosition } from '../Tooltip';
import { TimesCircleIcon } from '@patternfly/react-icons';
import styles from '@patternfly/patternfly/components/Chip/chip.css';
import GenerateId from '../../helpers/GenerateId/GenerateId';

class Chip extends React.Component {
  span = React.createRef();
  state = { isTooltipVisible: false };

  componentDidMount() {
    this.setState({
      isTooltipVisible: this.span.current && this.span.current.offsetWidth < this.span.current.scrollWidth
    });
  }

  renderOverflowChip = () => {
    const { children, className, onClick } = this.props;
    return (
      <div className={css(styles.chip, styles.modifiers.overflow, className)}>
        <ChipButton onClick={onClick}>
          <span className={css(styles.chipText)}>{children}</span>
        </ChipButton>
      </div>
    );
  };

  renderChip = randomId => {
    const { children, closeBtnAriaLabel, tooltipPosition, className, onClick } = this.props;
    if (this.state.isTooltipVisible) {
      return (
        <Tooltip position={tooltipPosition} content={children}>
          <li className={css(styles.chip, className)}>
            <span ref={this.span} className={css(styles.chipText)} id={randomId}>
              {children}
            </span>
            <ChipButton
              onClick={onClick}
              ariaLabel={closeBtnAriaLabel}
              id={`remove_${randomId}`}
              aria-labelledby={`remove_${randomId} ${randomId}`}
            >
              <TimesCircleIcon aria-hidden="true" />
            </ChipButton>
          </li>
        </Tooltip>
      );
    }
    return (
      <li className={css(styles.chip, className)}>
        <span ref={this.span} className={css(styles.chipText)} id={randomId}>
          {children}
        </span>
        <ChipButton
          onClick={onClick}
          ariaLabel={closeBtnAriaLabel}
          id={`remove_${randomId}`}
          aria-labelledby={`remove_${randomId} ${randomId}`}
        >
          <TimesCircleIcon aria-hidden="true" />
        </ChipButton>
      </li>
    );
  };

  render() {
    const { isOverflowChip } = this.props;
    return (
      <GenerateId>{randomId => (isOverflowChip ? this.renderOverflowChip() : this.renderChip(randomId))}</GenerateId>
    );
  }
}
Chip.propTypes = {
  /** Content rendered inside the chip text */
  children: PropTypes.node,
  /** Aria Label for close button */
  closeBtnAriaLabel: PropTypes.string,
  /** Additional classes added to the chip item */
  className: PropTypes.string,
  /** Flag indicating if the chip has overflow */
  isOverflowChip: PropTypes.bool,
  /** Function that is called when clicking on the chip button */
  onClick: PropTypes.func,
  /** Position of the tooltip which is displayed if text is longer */
  tooltipPosition: PropTypes.oneOf(Object.values(TooltipPosition))
};

Chip.defaultProps = {
  children: null,
  closeBtnAriaLabel: 'close',
  className: '',
  isOverflowChip: false,
  tooltipPosition: 'top'
};

export default Chip;
