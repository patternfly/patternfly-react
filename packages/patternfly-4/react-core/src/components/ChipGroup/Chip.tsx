import * as React from 'react';
import { css } from '@patternfly/react-styles';
import { ChipButton } from './ChipButton';
import { OneOf } from '../../helpers/typeUtils';
import { Tooltip, TooltipPosition } from '../Tooltip';
import { TimesCircleIcon } from '@patternfly/react-icons';
import styles from '@patternfly/react-styles/css/components/Chip/chip';
import GenerateId from '../../helpers/GenerateId/GenerateId';

export interface ChipProps extends React.HTMLProps<HTMLDivElement> {
  /** Content rendered inside the chip text */
  children?: React.ReactNode;
  /** Aria Label for close button */
  closeBtnAriaLabel?: string;
  /** Additional classes added to the chip item */
  className?: string; 
  /** Flag indicating if the chip has overflow */
  isOverflowChip?: boolean;
  /** Flag if chip is read only */
  isReadOnly?: boolean;
  /** Function that is called when clicking on the chip button */
  onClick?: (event: React.MouseEvent) => void
  /** Internal flag for which component will be used for chip */
  component?: React.ReactNode; 
  /** Position of the tooltip which is displayed if text is longer */
  tooltipPosition?: 'top' | 'bottom' | 'left' | 'right';
}

interface ChipState {
  isTooltipVisible: boolean;
}

export class Chip extends React.Component<ChipProps, ChipState> {
  constructor(props: ChipProps){
    super(props); 
    this.state = {
      isTooltipVisible: false
    }
  }
  span = React.createRef<HTMLSpanElement>(); 

  static defaultProps = {
    closeBtnAriaLabel: 'close',
    className: '',
    isOverflowChip: false,
    isReadOnly: false,
    tooltipPosition: 'top',
    onClick: () => undefined as any,
    component: 'div'
  }

  componentDidMount() {
    this.setState({
      isTooltipVisible: Boolean(this.span.current && 
        this.span.current.offsetWidth < this.span.current.scrollWidth)
    });
  }

  renderOverflowChip = () => {
    const { children, className, onClick } = this.props;
    const Component = this.props.component as any;
    return (
      <Component className={css(styles.chip, styles.modifiers.overflow, className)}>
        <ChipButton onClick={onClick}>
          <span className={css(styles.chipText)}>{children}</span>
        </ChipButton>
      </Component>
    );
  };

  renderChip = (randomId: string) => {
    const {
      children,
      closeBtnAriaLabel,
      tooltipPosition,
      className,
      onClick,
      isReadOnly
    } = this.props;
    const Component = this.props.component as any;
    if (this.state.isTooltipVisible) {
      return (
        <Tooltip position={tooltipPosition} content={children}>
          <Component className={css(styles.chip, isReadOnly && styles.modifiers.readOnly, className)}>
            <span ref={this.span} className={css(styles.chipText)} id={randomId}>
              {children}
            </span>
            {!isReadOnly && (
              <ChipButton
                onClick={onClick}
                ariaLabel={closeBtnAriaLabel}
                id={`remove_${randomId}`}
                aria-labelledby={`remove_${randomId} ${randomId}`}
              >
                <TimesCircleIcon aria-hidden="true" />
              </ChipButton>
            )}
          </Component>
        </Tooltip>
      );
    }
    return (
      <Component className={css(styles.chip, isReadOnly && styles.modifiers.readOnly, className)}>
        <span ref={this.span} className={css(styles.chipText)} id={randomId}>
          {children}
        </span>
        {!isReadOnly && (
          <ChipButton
            onClick={onClick}
            ariaLabel={closeBtnAriaLabel}
            id={`remove_${randomId}`}
            aria-labelledby={`remove_${randomId} ${randomId}`}
          >
            <TimesCircleIcon aria-hidden="true" />
          </ChipButton>
        )}
      </Component>
    );
  };
  
  render(){
    const { isOverflowChip } = this.props;
    return (
      <GenerateId>{randomId => (isOverflowChip ? this.renderOverflowChip() : this.renderChip(randomId))}</GenerateId>
    );
  }
};
