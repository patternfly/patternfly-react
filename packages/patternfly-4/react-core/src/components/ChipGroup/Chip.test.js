import React from 'react';
import { shallow } from 'enzyme';
import ChipButton from './ChipButton';
import Chip from './Chip';

test('ChipButton', () => {
  const view = shallow(
    <ChipButton id="my-chip-button" className="chip-bttn-cls">
      <b>Close</b>
    </ChipButton>
  );
  expect(view).toMatchSnapshot();
});

describe('Chip', () => {
  test('overflow', () => {
    const view = shallow(
      <Chip className="my-chp-cls" isOverflowChip>
        4 more
      </Chip>
    );
    expect(view).toMatchSnapshot();
  });

  test('closable', () => {
    const view = shallow(
      <Chip className="my-chp-cls" id="chip_one">
        Chip
      </Chip>
    );
    expect(view).toMatchSnapshot();
  });

  test('closable with tooltip', () => {
    const view = shallow(
      <Chip className="my-chp-cls" id="chip_one">
        1234567890123456789
      </Chip>
    );
    expect(view).toMatchSnapshot();
  });
});
