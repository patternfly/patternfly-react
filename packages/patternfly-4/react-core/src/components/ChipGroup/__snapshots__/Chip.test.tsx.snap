// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip closable 1`] = `
<Component
  className="my-chp-cls"
  id="chip_one"
>
  <ComponentWithOuia
    component={[Function]}
    componentProps={
      Object {
        "children": "Chip",
        "className": "my-chp-cls",
        "id": "chip_one",
      }
    }
    consumerContext={null}
  >
    <Chip
      className="my-chp-cls"
      closeBtnAriaLabel="close"
      component="div"
      id="chip_one"
      isOverflowChip={false}
      isReadOnly={false}
      onClick={[Function]}
      ouiaContext={
        Object {
          "isOuia": false,
          "ouiaId": null,
        }
      }
      tooltipPosition="top"
    >
      <GenerateId
        prefix="pf-random-id-"
      >
        <div
          className="pf-c-chip my-chp-cls"
        >
          <span
            className="pf-c-chip__text"
            id="pf-random-id-1"
          >
            Chip
          </span>
          <Component
            aria-labelledby="remove_pf-random-id-1 pf-random-id-1"
            ariaLabel="close"
            id="remove_pf-random-id-1"
            onClick={[Function]}
          >
            <Component
              aria-label="close"
              aria-labelledby="remove_pf-random-id-1 pf-random-id-1"
              className=""
              id="remove_pf-random-id-1"
              onClick={[Function]}
              variant="plain"
            >
              <ComponentWithOuia
                component={[Function]}
                componentProps={
                  Object {
                    "aria-label": "close",
                    "aria-labelledby": "remove_pf-random-id-1 pf-random-id-1",
                    "children": <TimesCircleIcon
                      aria-hidden="true"
                      color="currentColor"
                      noVerticalAlign={false}
                      size="sm"
                      title={null}
                    />,
                    "className": "",
                    "id": "remove_pf-random-id-1",
                    "onClick": [Function],
                    "variant": "plain",
                  }
                }
                consumerContext={
                  Object {
                    "isOuia": false,
                    "ouiaId": null,
                  }
                }
              >
                <Button
                  aria-label="close"
                  aria-labelledby="remove_pf-random-id-1 pf-random-id-1"
                  className=""
                  id="remove_pf-random-id-1"
                  onClick={[Function]}
                  ouiaContext={
                    Object {
                      "isOuia": false,
                      "ouiaId": null,
                    }
                  }
                  variant="plain"
                >
                  <button
                    aria-disabled={null}
                    aria-label="close"
                    aria-labelledby="remove_pf-random-id-1 pf-random-id-1"
                    className="pf-c-button pf-m-plain"
                    disabled={false}
                    id="remove_pf-random-id-1"
                    onClick={[Function]}
                    tabIndex={null}
                    type="button"
                  >
                    <TimesCircleIcon
                      aria-hidden="true"
                      color="currentColor"
                      noVerticalAlign={false}
                      size="sm"
                      title={null}
                    >
                      <svg
                        aria-hidden="true"
                        aria-labelledby={null}
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style={
                          Object {
                            "verticalAlign": "-0.125em",
                          }
                        }
                        viewBox="0 0 512 512"
                        width="1em"
                      >
                        <path
                          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
                          transform=""
                        />
                      </svg>
                    </TimesCircleIcon>
                  </button>
                </Button>
              </ComponentWithOuia>
            </Component>
          </Component>
        </div>
      </GenerateId>
    </Chip>
  </ComponentWithOuia>
</Component>
`;

exports[`Chip closable with tooltip 1`] = `
<Component
  className="my-chp-cls"
  id="chip_one"
>
  <ComponentWithOuia
    component={[Function]}
    componentProps={
      Object {
        "children": "1234567890123456789",
        "className": "my-chp-cls",
        "id": "chip_one",
      }
    }
    consumerContext={null}
  >
    <Chip
      className="my-chp-cls"
      closeBtnAriaLabel="close"
      component="div"
      id="chip_one"
      isOverflowChip={false}
      isReadOnly={false}
      onClick={[Function]}
      ouiaContext={
        Object {
          "isOuia": false,
          "ouiaId": null,
        }
      }
      tooltipPosition="top"
    >
      <GenerateId
        prefix="pf-random-id-"
      >
        <div
          className="pf-c-chip my-chp-cls"
        >
          <span
            className="pf-c-chip__text"
            id="pf-random-id-2"
          >
            1234567890123456789
          </span>
          <Component
            aria-labelledby="remove_pf-random-id-2 pf-random-id-2"
            ariaLabel="close"
            id="remove_pf-random-id-2"
            onClick={[Function]}
          >
            <Component
              aria-label="close"
              aria-labelledby="remove_pf-random-id-2 pf-random-id-2"
              className=""
              id="remove_pf-random-id-2"
              onClick={[Function]}
              variant="plain"
            >
              <ComponentWithOuia
                component={[Function]}
                componentProps={
                  Object {
                    "aria-label": "close",
                    "aria-labelledby": "remove_pf-random-id-2 pf-random-id-2",
                    "children": <TimesCircleIcon
                      aria-hidden="true"
                      color="currentColor"
                      noVerticalAlign={false}
                      size="sm"
                      title={null}
                    />,
                    "className": "",
                    "id": "remove_pf-random-id-2",
                    "onClick": [Function],
                    "variant": "plain",
                  }
                }
                consumerContext={
                  Object {
                    "isOuia": false,
                    "ouiaId": null,
                  }
                }
              >
                <Button
                  aria-label="close"
                  aria-labelledby="remove_pf-random-id-2 pf-random-id-2"
                  className=""
                  id="remove_pf-random-id-2"
                  onClick={[Function]}
                  ouiaContext={
                    Object {
                      "isOuia": false,
                      "ouiaId": null,
                    }
                  }
                  variant="plain"
                >
                  <button
                    aria-disabled={null}
                    aria-label="close"
                    aria-labelledby="remove_pf-random-id-2 pf-random-id-2"
                    className="pf-c-button pf-m-plain"
                    disabled={false}
                    id="remove_pf-random-id-2"
                    onClick={[Function]}
                    tabIndex={null}
                    type="button"
                  >
                    <TimesCircleIcon
                      aria-hidden="true"
                      color="currentColor"
                      noVerticalAlign={false}
                      size="sm"
                      title={null}
                    >
                      <svg
                        aria-hidden="true"
                        aria-labelledby={null}
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style={
                          Object {
                            "verticalAlign": "-0.125em",
                          }
                        }
                        viewBox="0 0 512 512"
                        width="1em"
                      >
                        <path
                          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
                          transform=""
                        />
                      </svg>
                    </TimesCircleIcon>
                  </button>
                </Button>
              </ComponentWithOuia>
            </Component>
          </Component>
        </div>
      </GenerateId>
    </Chip>
  </ComponentWithOuia>
</Component>
`;

exports[`Chip overflow 1`] = `
<Component
  className="my-chp-cls"
  isOverflowChip={true}
>
  <ComponentWithOuia
    component={[Function]}
    componentProps={
      Object {
        "children": "4 more",
        "className": "my-chp-cls",
        "isOverflowChip": true,
      }
    }
    consumerContext={null}
  >
    <Chip
      className="my-chp-cls"
      closeBtnAriaLabel="close"
      component="div"
      isOverflowChip={true}
      isReadOnly={false}
      onClick={[Function]}
      ouiaContext={
        Object {
          "isOuia": false,
          "ouiaId": null,
        }
      }
      tooltipPosition="top"
    >
      <GenerateId
        prefix="pf-random-id-"
      >
        <div
          className="pf-c-chip pf-m-overflow my-chp-cls"
        >
          <Component
            onClick={[Function]}
          >
            <Component
              aria-label="close"
              className=""
              onClick={[Function]}
              variant="plain"
            >
              <ComponentWithOuia
                component={[Function]}
                componentProps={
                  Object {
                    "aria-label": "close",
                    "children": <span
                      className="pf-c-chip__text"
                    >
                      4 more
                    </span>,
                    "className": "",
                    "onClick": [Function],
                    "variant": "plain",
                  }
                }
                consumerContext={
                  Object {
                    "isOuia": false,
                    "ouiaId": null,
                  }
                }
              >
                <Button
                  aria-label="close"
                  className=""
                  onClick={[Function]}
                  ouiaContext={
                    Object {
                      "isOuia": false,
                      "ouiaId": null,
                    }
                  }
                  variant="plain"
                >
                  <button
                    aria-disabled={null}
                    aria-label="close"
                    className="pf-c-button pf-m-plain"
                    disabled={false}
                    onClick={[Function]}
                    tabIndex={null}
                    type="button"
                  >
                    <span
                      className="pf-c-chip__text"
                    >
                      4 more
                    </span>
                  </button>
                </Button>
              </ComponentWithOuia>
            </Component>
          </Component>
        </div>
      </GenerateId>
    </Chip>
  </ComponentWithOuia>
</Component>
`;

exports[`Chip readonly 1`] = `
<Component
  className="my-chp-cls"
  isReadOnly={true}
>
  <ComponentWithOuia
    component={[Function]}
    componentProps={
      Object {
        "children": "4 more",
        "className": "my-chp-cls",
        "isReadOnly": true,
      }
    }
    consumerContext={null}
  >
    <Chip
      className="my-chp-cls"
      closeBtnAriaLabel="close"
      component="div"
      isOverflowChip={false}
      isReadOnly={true}
      onClick={[Function]}
      ouiaContext={
        Object {
          "isOuia": false,
          "ouiaId": null,
        }
      }
      tooltipPosition="top"
    >
      <GenerateId
        prefix="pf-random-id-"
      >
        <div
          className="pf-c-chip pf-m-read-only my-chp-cls"
        >
          <span
            className="pf-c-chip__text"
            id="pf-random-id-3"
          >
            4 more
          </span>
        </div>
      </GenerateId>
    </Chip>
  </ComponentWithOuia>
</Component>
`;

exports[`ChipButton 1`] = `
<Component
  className="chip-bttn-cls"
  id="my-chip-button"
>
  <Component
    aria-label="close"
    className="chip-bttn-cls"
    id="my-chip-button"
    onClick={[Function]}
    variant="plain"
  >
    <ComponentWithOuia
      component={[Function]}
      componentProps={
        Object {
          "aria-label": "close",
          "children": <b>
            Close
          </b>,
          "className": "chip-bttn-cls",
          "id": "my-chip-button",
          "onClick": [Function],
          "variant": "plain",
        }
      }
      consumerContext={null}
    >
      <Button
        aria-label="close"
        className="chip-bttn-cls"
        id="my-chip-button"
        onClick={[Function]}
        ouiaContext={
          Object {
            "isOuia": false,
            "ouiaId": null,
          }
        }
        variant="plain"
      >
        <button
          aria-disabled={null}
          aria-label="close"
          className="pf-c-button pf-m-plain chip-bttn-cls"
          disabled={false}
          id="my-chip-button"
          onClick={[Function]}
          tabIndex={null}
          type="button"
        >
          <b>
            Close
          </b>
        </button>
      </Button>
    </ComponentWithOuia>
  </Component>
</Component>
`;
